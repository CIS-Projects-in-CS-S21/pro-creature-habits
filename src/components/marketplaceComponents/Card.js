import React from "react";
import {Image, Text, View, StyleSheet} from "react-native";
import {ItemInventory} from "../ItemInventory";
import {useSelector} from "react-redux";
import {Ionicons} from "@expo/vector-icons";

const styles = StyleSheet.create({
	cardContainer: {
		borderWidth: 3,
		borderColor: 'white',
		padding: 12,
		backgroundColor: 'rgba(49,69,194,0.7)',
		shadowOffset: {width: 0, height: 1},
		shadowOpacity: 0.8,
		shadowRadius: 3,
		elevation: 11,
		borderRadius: 5
	},
	cardFooter: {
		flexDirection: 'row',
		justifyContent: 'space-evenly',
		paddingLeft: 5,
		paddingRight: 5,
		marginTop: 5,
		borderTopWidth: 2,
		borderTopColor: 'darkgrey'
	},
	cardFooterText: {
		fontSize: 20,
		color: 'white',
		marginTop: 5
	},
	coin: {
		width: 25,
		height: 25,
		marginTop: 5
	},
	itemImage: {
		width:75,
		height:75,
		shadowOffset: {width: 0, height: 1},
		shadowOpacity: 0.8,
		shadowRadius: 3,
	}
});

const Card = ({item}) => {
	const itemsBought = useSelector(state=>state.itemsBought);

	return (
		<View>
			{!itemsBought.includes(item) ? (
				<View style={styles.cardContainer}>
					<Image style={styles.itemImage} source={ItemInventory[item].uri}/>
					<View style={styles.cardFooter}>
						<Text style={styles.cardFooterText}>{ItemInventory[item].cost}</Text>
						<Image style={styles.coin} source={require('../../test_images/coin.png')}/>
					</View>
				</View>
			) : (
				<View style={[styles.cardContainer, {backgroundColor: '#2a2a2a'}]}>
					<Image style={styles.itemImage} source={ItemInventory[item].uri}/>
					<View style={styles.cardFooter}>
						<Ionicons name="checkmark" color="white" size={28}/>
					</View>
				</View>
			)}
		</View>
	)
}

export default Card;
