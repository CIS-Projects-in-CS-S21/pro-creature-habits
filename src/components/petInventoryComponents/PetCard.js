import React from "react";
import {Image, Text, View, StyleSheet} from "react-native";
import {ItemInventory} from "../ItemInventory";

const styles = StyleSheet.create({
	cardContainer: {
		borderWidth: 2,
		padding: 8,
		backgroundColor: 'rgba(49,69,194,0.7)',
		shadowOffset: {width: 0, height: 5},
		shadowOpacity: 0.8,
		shadowRadius: 3,
		elevation: 11
	},
	circle: {
       width: 20,
       height: 20,
       borderRadius: 10,
       color: 'white',
       borderWidth: 1,
       borderColor: 'white',
       borderStyle: 'solid',
       textAlign: "center",
       fontWeight: 'bold'
    },
	itemImage: {
		width:75,
		height:75,
		shadowOffset: {width: 0, height: 1},
		shadowOpacity: 0.8,
		shadowRadius: 3,
	}
});

const PetCard = ({item}) => {
	return (
		<View style={styles.cardContainer}>
		    <View style={{flexDirection: "row", justifyContent: "flex-end"}}>
		     {(!item.wear) ? <Text style={styles.circle}> {item.bought} </Text> : <Text style={styles.circle}> âˆš </Text>}
		    </View>
			<Image style={styles.itemImage} source={item.uri}/>
		</View>
	)
}

export default PetCard;
