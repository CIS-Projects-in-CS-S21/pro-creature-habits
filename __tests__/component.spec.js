import * as React from 'react';
import renderer from 'react-test-renderer';
import PetCard from '../src/components/petInventoryComponents/PetCard';
import Card from '../src/components/marketplaceComponents/Card'
import * as enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { Provider } from 'react-redux';


enzyme.configure({ adapter: new Adapter() });


describe('<PetCard />', () => {

    /*it(`WeatherOverlay renders Correctly on Load`, () => {

        const petCardComponent = renderer.create(<PetCard />).toJSON()
        expect(weatherOverlayComponent).toMatchSnapshot();
        });*/

    it('petCard is not null if item exists', () => {

        const petCardComponent = renderer.create(<PetCard item="shoe" />).toJSON();
        expect(petCardComponent).not.toBeNull();
    })

    it("petCard not null", () => {

        expect(PetCard).toBeTruthy();
    })

    // it('checks to see if correct icon shown', () => {
    //     const wrapper = shallow(
    //     <PetCard item="shoe" />
    //     );

    //     const render = wrapper.dive();
    //     icons = render.find('Ionicons');
    //     expect(icons.name).toEqual("clothing")
    // });

    it('petCard snapshot renders',()=>{
        const petCardComponent = renderer.create(<PetCard item="shoe"/>).toJSON()
        expect(petCardComponent).toMatchSnapshot();
    });

    const mockProp = {
        wear:true, category:'food', bought:1, uri:require('../src/test_images/burger.png'), 
    }
    // const initialState = { output: "null" }
    // const mockStore = configureStore()

    it(`PetCard renders correctly`, () => {
        
        // store = mockStore(initialState)
        const comp = <Provider><PetCard event={mockProp} /></Provider>
        const tree = renderer.create().toJSON(comp);

        expect(tree).toMatchSnapshot();
    });

})

// describe('<Card />', ()=>{
//     it('Card snapshot renders',()=>{
//         const CardComponent = renderer.create(<Provider><Card item="burger" Ionicons={'checkmark'}/></Provider>).toJSON()
//         expect(CardComponent).toMatchSnapshot();
//     });
// })