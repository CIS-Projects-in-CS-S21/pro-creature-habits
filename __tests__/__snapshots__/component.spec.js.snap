// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PetCard /> PetCard renders correctly 1`] = `null`;

exports[`<PetCard /> petCard snapshot renders 1`] = `
<View>
  <View
    style={
      Object {
        "backgroundColor": "rgba(49,69,194,0.7)",
        "borderColor": "white",
        "borderRadius": 5,
        "borderWidth": 3,
        "elevation": 11,
        "padding": 10,
        "shadowOffset": Object {
          "height": 1,
          "width": 0,
        },
        "shadowOpacity": 0.8,
        "shadowRadius": 3,
      }
    }
  >
    <Image
      style={
        Object {
          "height": 75,
          "shadowOffset": Object {
            "height": 1,
            "width": 0,
          },
          "shadowOpacity": 0.8,
          "shadowRadius": 3,
          "width": 75,
        }
      }
    />
  </View>
</View>
`;
